### ISSUE TITLE
AxeBuilder violation results are not consistent in @axe-core/playwright

### ISSUE THREAD

**Issue by youvegotnigel** [dequelabs/axe-core-npm#1085]
**Describe the bug**
I was experimenting with axe-core and playwright for accessibility testing in our organization and noticed this stage behavior and wanted to figure out what was going on here.

**To Reproduce**
1. When I execute below code in playwright I get only one violation for **{"description": "Ensures that lists are structured correctly"**

`
  test("CDC Dengue page should have only two WCAG A or AA violations", async ({page}, testInfo) => {
    await page.goto(`https://www.cdc.gov/dengue/about/index.html`);

    // wait for content to load
    await page.waitForLoadState("domcontentloaded");

    const accessibilityScanResults = await new AxeBuilder({ page })
      .withTags(["wcag2a", "wcag2aa"])
      .analyze();

    await testInfo.attach("accessibility-scan-results", {
      body: JSON.stringify(accessibilityScanResults, null, 2),
      contentType: "application/json",
    });

    expect(accessibilityScanResults.violations).toHaveLength(2);

  });`

2. But when I execute the same test but only added `withRules(["skip-link"])`, I still only get one violation and this time it is **[{"description": "Ensure all skip links have a focusable target",**

`test("CDC Dengue page should have only two WCAG A or AA violations with skip link rule", async ({page}, testInfo) => {
    await page.goto(`https://www.cdc.gov/dengue/about/index.html`);

    // wait for content to load
    await page.waitForLoadState("domcontentloaded");

    const accessibilityScanResults = await new AxeBuilder({ page })
      .withTags(["wcag2a", "wcag2aa"])
      .withRules([
         "skip-link"
      ])
      .analyze();

    await testInfo.attach("accessibility-scan-results", {
      body: JSON.stringify(accessibilityScanResults, null, 2),
      contentType: "application/json",
    });

    expect(accessibilityScanResults.violations).toHaveLength(2);
    
  });
`

**Expected behavior**
In the second test I should have got **two violations**.

1. Ensures that lists are structured correctly
2. Ensure all skip links have a focusable target

**Environment :**

- OS: Windows 10
- Playwright version: 1.45.0
- axe-core/playwright version: 4.9.1

**Additional context**

- Full error for test no 1:
==========================================================================
Error: expect(received).toHaveLength(expected)

    Expected length: 2
    Received length: 1
    Received array:  [{"description": "Ensures that lists are structured correctly", "help": "<ul> and <ol> must only directly contain <li>, <script> or <template> elements", "helpUrl": "https://dequeuniversity.com/rules/axe/4.9/list?application=playwright", "id": "list", "impact": "serious", "nodes": [{"all": [], "any": [], "failureSummary": "Fix all of the following:
      List element has direct children that are not allowed: a", "html": "<ul>", "impact": "serious", "none": [{"data": {"values": "a"}, "id": "only-listitems", "impact": "serious", "message": "List element has direct children that are not allowed: a", "relatedNodes": [{"html": "<a class=\"more-link\" href=\"#\">Show More</a>", "target": [".more-link"]}]}], "target": [".card-body > ul"]}], "tags": ["cat.structure", "wcag2a", "wcag131", "EN-301-549", "EN-9.1.3.1"]}]
==========================================================================

- Full error for test no 2:

==========================================================================
Error: expect(received).toHaveLength(expected)

    Expected length: 2
    Received length: 1
    Received array:  [{"description": "Ensure all skip links have a focusable target", "help": "The skip-link target should exist and be focusable", "helpUrl": "https://dequeuniversity.com/rules/axe/4.9/skip-link?application=playwright", "id": "skip-link", "impact": "moderate", "nodes": [{"all": [], "any": [{"data": null, "id": "skip-link", "impact": "moderate", "message": "No skip link target", "relatedNodes": []}], "failureSummary": "Fix any of the following:
      No skip link target", "html": "<a class=\"visually-hidden-focusable\" href=\"#cdc-search\">Skip directly to search</a>", "impact": "moderate", "none": [], "target": ["a[href$=\"#cdc-search\"]"]}], "tags": ["cat.keyboard", "best-practice"]}]

==========================================================================

**Comment by Zidious** [comment]
Hey @youvegotnigel,

Just confirming the URL under test is https://www.cdc.gov/dengue/about/index.html%60? One thing to note here, when utilising `withTags(["wcag2a", "wcag2aa"])` this will run all the rules that fall under `wcag2a` and `wcag2aa`. However, if chained with `withRules('skip-link')` it will _only_ run that one rule against the page under test.

**Comment by youvegotnigel** [comment]
Hi @Zidious 
Thank u for quick response.
Actually, the URL is "https://www.cdc.gov/dengue/about/index.html"

BTW I was referring to this on YouTube: https://www.youtube.com/watch?v=cs5-Kk9nQDA&t=410s

Is it not possible to check for both "wcag2a" and "wcag2aa" a11y specification and opt-in selected a11y checks as well?

**Comment by Zidious** [comment]
Ah I see. I suspect what you want to use the [`options(..)`](https://github.com/dequelabs/axe-core-npm/tree/develop/packages/playwright#axebuilderoptionsoptions-axerunoptions) API if you want to run all rules relating to WCAG2A and WCAG2AA but also run the skip-link rule, example usage: 

```ts
import puppeteer from "puppeteer";
import { AxePuppeteer } from "@axe-core/puppeteer";

(async () => {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();
  await page.goto("https://www.cdc.gov/dengue/about/index.html");

  const axeBuilder = new AxePuppeteer(page).options({
    runOnly: {
      type: "tag",
      values: ["wcag2a", "wcag2aa"],
    },
    rules: {
      "skip-link": { enabled: true },
    },
  });

  const { violations } = await axeBuilder.analyze();

  // Do something with the violations
})();
```

**Comment by youvegotnigel** [comment]
Thank you very much @Zidious. 
This works as I expected in playwright too.
